File: 2535. Difference Between Element Sum and Digit Sum of an Array.java
class Solution {
    public int differenceOfSum(int[] nums) {
        int sum = 0, digitSum = 0;

        for(int i = 0; i < nums.length; i++){
            sum += nums[i];
            int temp = nums[i];
            while(temp > 0){
                int t = temp % 10;
                digitSum += t;
                temp /= 10;
            }
        }

        return Math.abs(digitSum - sum);
    }
}

File: 2535. Difference Between Element Sum and Digit Sum of an Array.ts
function differenceOfSum(nums: number[]): number {
    let sumOfDigits = 0;
    let sumOfElement = 0;
    
    for(const num of nums){
        sumOfElement += num;
        for(const digit of getDigits(num)){
            sumOfDigits += digit;
        }
    }

    return Math.abs(sumOfDigits - sumOfElement);
};

function getDigits(num: number){
    const listOfDigits = [];
    while(num>0){
        const t = num % 10;
        listOfDigits.push(t);
        num = Math.floor(num/10);
    }

    return listOfDigits;
};
